sources/osgShader/Compiler.js

```diff
@@ -775,6 +783,34 @@ define( [
 
         },
 
+        createShadowTextureInputVarying: function ( shadowTexture, inputs, vertexWorld, tUnit ) {
+            var shadowTexSamplerName = 'ShadowTexture' + tUnit;
+
+            // per texture uniforms
+            var shadowTextureUniforms = this.getOrCreateTextureStateAttributeUniforms( shadowTexture, 'shadowTexture', tUnit );
+
+            // UGLY REMOVAL
+            // TODO: create a specific texture getuniform for compiler
+            var id = shadowTextureUniforms[ 'shadowTexture' + shadowTexSamplerName ].getID();
+            shadowTextureUniforms[ 'shadowTexture' + shadowTexSamplerName ] = undefined;
+            this._variables[ shadowTexSamplerName ] = undefined;
+            this._activeNodeList[ id ] = undefined;
+            delete this._activeNodeList[ id ];
+            // end UGLY REMOVAL
+
+            inputs = MACROUTILS.objectMix( inputs, shadowTextureUniforms );
```

Comment:
dont override arguments variable, use a different name. not sure it works when strict is used. To avoid any doubt use a different variable name